C-Plus-Plus/graphs/BellmanFord.cpp

```diff
@@ -0,0 +1,75 @@
+
+/*Input:
+First line contains two space separated integers,(N,M) 
+N- no of vertices, M- no of edges.
+Then M lines follow, each line has 3 space separated integers ui ,vi ,wi which denotes edge from vertex ui to vi with weight wi.
+Find shortest distance from vertex 1 to all other vertices.
+
+
+TEST CASE
+5 5
+1 2 5
+1 3 2
+3 4 1
+1 4 6
+3 5 5
+
+OUTPUT
+5 2 3 7 */
+
+
+#include<bits/stdc++.h>
+using namespace std;
+#define tezz_chal ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
+#define ll long long int
+
+vector<int>edges[1000001];
```

Comment:
Why do we need to declare variables globally?

Reply:
Are you talking about the edges vector and distance array?
Well declaring them globally makes it easier, we don't have to pass all the variables to all the functions since they are globally available.